src/quacc/runners/ase.py

```diff
@@ -194,6 +195,14 @@ def run_opt(
 
     # Run calculation
     with traj, optimizer(atoms, **optimizer_kwargs) as dyn:
+        if optimizer_patch is not None:
+
+            class PatchedOptimizer(optimizer_patch, dyn.__class__):
+                pass
+
+            dyn.__class__ = PatchedOptimizer
+            dyn.__init__(**optimizer_kwargs)
```

Comment:
Hmm, why not simply define a custom optimizer in this case then? Why patch an existing optimizer in this way rather than have it be a new `optimizer` class being called above?